src/resource/utils/fetch-resource.js

```diff
@@ -119,6 +118,9 @@ export default async function fetchResource(url, parsedUrl) {
       response,
     };
   } catch (e) {
-    return Errors.badUrl;
+    return {
+      error: true,
+      messages: e.message,
```

Comment:
Let's make this `message` instead of `messages` — `messages` makes it sound like it's an array.